### ISSUE TITLE
zconf.hash.c:167:1: error: conflicting types for ‘kconf_id_lookup’

### ISSUE THREAD

**Issue by daxim** [McArcady/lnp-forge#4]
One of the depedencies fails to build:

```
› git rev-parse HEAD
f05f8aad15b9987ac13731421fced33feff075ce


› ./configure --prefix=/tmp/lnp
configure: loading site script /usr/share/site/x86_64-unknown-linux-gnu
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for a sed that does not truncate output... /usr/bin/sed
checking whether sed understands -r -i -e... yes
checking whether ln -s works... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for ranlib... ranlib
checking for gobjcopy... no
checking for objcopy... objcopy
checking for absolute path to objcopy... /usr/bin/objcopy
checking for gobjdump... no
checking for objdump... objdump
checking for absolute path to objdump... /usr/bin/objdump
checking for greadelf... no
checking for readelf... readelf
checking for absolute path to readelf... /usr/bin/readelf
checking for gperf... gperf
checking for absolute path to gperf... /usr/bin/gperf
checking for bison... bison
checking for flex... flex
checking for makeinfo... makeinfo
checking for cut... cut
checking for stat... stat
checking for readlink... readlink
checking for wget... wget
checking for tar... tar
checking for gzip... gzip
checking for bzip2... bzip2
checking for help2man... help2man
checking for gpatch... no
checking for patch... patch
checking for absolute path to patch... /usr/bin/patch
checking for bash >= 3.1... /bin/bash
checking for GNU awk... /usr/bin/gawk
checking for GNU make >= 3.80... /usr/bin/gmake
checking whether /usr/bin/gmake sets $(MAKE)... yes
checking for make 3.81... no
checking for cmake... /usr/bin/cmake
checking for qmake-qt5... /usr/bin/qmake-qt5
checking for GNU libtoolize >= 1.5.26... /usr/bin/libtoolize
checking for GNU automake >= 1.10... /usr/bin/automake
checking for tclsh... /usr/bin/tclsh
checking for wish... /usr/bin/wish
checking tcl version... 8.6
checking for python... /usr/bin/python
checking python module: Tkinter... yes
checking for java... /usr/bin/java
checking for java 1.8... yes
checking for git... /usr/bin/git
checking for hg... /usr/bin/hg
checking for perl... perl
checking for perl module XML::LibXML... ok
checking for perl module XML::LibXSLT... ok
checking for library containing IMG_Load... no
checking for library containing TTF_OpenFont... no
checking for library containing gzopen... no
configure: WARNING: Some 32-bits dependencies are missing, DwarfFortress < 0.43.05 will not be available
checking for IMG_Load in -l:libSDL_image-1.2.so.0... yes
checking for TTF_OpenFont in -l:libSDL_ttf-2.0.so.0... yes
checking for gzopen in -lz... yes
checking for xz... xz
checking for inline... inline
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible malloc... yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible realloc... yes
checking for size_t... yes
checking for working alloca.h... yes
checking for alloca... yes
checking libintl.h usability... yes
checking libintl.h presence... yes
checking for libintl.h... yes
checking whether gettext is declared... yes
checking for library containing bindtextdomain... none required
checking ncurses/ncurses.h usability... yes
checking ncurses/ncurses.h presence... yes
checking for ncurses/ncurses.h... yes
checking for library containing initscr... -lncursesw
checking for library containing tgetent... none required
configure: creating ./config.status
config.status: creating Makefile


› make
  SED    'lnp-forge'
  SED    'scripts/lnp-forge.sh'
  SED    'scripts/showTuple.sh'
  GEN    'config/configure.in'
  GEN    'paths.mk'
  GEN    'paths.sh'
  DEP    'nconf.gui.dep'
  DEP    'nconf.dep'
  DEP    'lxdialog/yesno.dep'
  DEP    'lxdialog/util.dep'
  DEP    'lxdialog/textbox.dep'
  DEP    'lxdialog/menubox.dep'
  DEP    'lxdialog/inputbox.dep'
  DEP    'lxdialog/checklist.dep'
  DEP    'mconf.dep'
  DEP    'conf.dep'
  BISON  'zconf.tab.c'
  GPERF  'zconf.hash.c'
  LEX    'zconf.lex.c'
  DEP    'zconf.tab.dep'
  CC     'zconf.tab.o'
In file included from zconf.tab.c:212:0:
zconf.hash.c:167:1: error: conflicting types for ‘kconf_id_lookup’
 kconf_id_lookup (register const char *str, register size_t len)
 ^~~~~~~~~~~~~~~
zconf.hash.c:34:31: note: previous declaration of ‘kconf_id_lookup’ was here
 static const struct kconf_id *kconf_id_lookup(register const char *str, register unsigned int len);
                               ^~~~~~~~~~~~~~~
make[2]: *** [Makefile:89: zconf.tab.o] Fehler 1
make[1]: *** [Makefile:171: build-lib-kconfig] Fehler 2
make: *** [Makefile:132: build] Fehler 2
```

**Comment by McArcady** [comment]
What version of package 'gperf' are you using?
gperf-3.1 has been known to cause problems (e.g [https://github.com/pfalcon/esp-open-sdk/issues/243](url))
If your gperf was upgraded to 3.1, could you try downgrading it to 3.0.4 for instance (or the last available version <3.1 in your distro)

**Comment by daxim** [comment]
Yes, it's 3.1.

I'm on a rolling distro and got lucky that there is still an unofficial build of 3.0.4 hanging around. Downgrading works, i.e. I can compile and use the program.

Mind this is only a non-sustainable work-around, and not a solution. The build remains broken for some users.

If upstream does not fix their dependency problem real soon now, you need to implement a different menu system, perhaps <https://techbase.kde.org/Development/Tutorials/Shell_Scripting_with_KDE_Dialogs#Menu_and_Selection_Dialogs>.

**Comment by McArcady** [comment]
I pushed a fix on branch master (47ff9af714)
Could you give it a try with gperf-3.1?

**Comment by daxim** [comment]
confirmed fixed
